core/src/main/java/org/infinispan/cache/impl/DecoratedCache.java

```diff
@@ -476,22 +577,25 @@ public void removeGroup(String groupName) {
 
    @Override
    public V putIfAbsent(K key, V value) {
-      return cacheImplementation.putIfAbsent(key, value, cacheImplementation.defaultMetadata, flags);
+      return putIfAbsent(key, value, cacheImplementation.defaultMetadata);
    }
 
    @Override
    public boolean remove(Object key, Object value) {
-      return cacheImplementation.remove(key, value, flags);
+      cacheImplementation.assertKeyValueNotNull(key, value);
+      InvocationContext ctx = cacheImplementation.getInvocationContextWithImplicitTransaction(false, 1);
```

Comment:
Hrmm weird looking at the diff again part of my changes were reverted... I will have to fix these up.